cpp/dolfinx/common/subsystem.h

```diff
@@ -27,11 +29,11 @@ void init_logging(int argc, char* argv[]);
 
 /// Initialize PETSc (and SLEPc, if configured) without command-line
 /// arguments
-void init_petsc();
+void init_petsc(MPI_Comm comm = MPI_COMM_WORLD);
```

Comment:
Would it be better to make the default `MPI_COMM_NULL` == let PETSc manage the MPI initialisation?

Reply:
I'm not sure which is best. If `PETSC_COMM_WORLD` == `MPI_COMM_WORLD` then it's just the same as letting PETSc manage the initialisation. 

If however, you want people to be able to let `init_petsc` do MPI initialisation when people aren't specifying explicit communicators, but require people to initialise MPI manually prior to `init_petsc` if they are specifying communicators, this would be the way to go.

I guess this is a design decision that is probably best not made by me.